src/sentry/release_health/metrics.py

```diff
@@ -122,7 +123,83 @@ def _model_environment_ids_to_environment_names(
 
 
 class MetricsReleaseHealthBackend(ReleaseHealthBackend):
-    """Gets release health results from the metrics dataset"""
+    """
+    Implementation of the ReleaseHealthBackend using the MetricsLayer API
+    """
+
+    @staticmethod
+    def _get_org_id(project_ids: Sequence[int]) -> int:
+        return MetricsReleaseHealthBackend._get_projects_and_org_id(project_ids)[1]
+
+    @staticmethod
+    def _get_projects(project_ids: Sequence[int]) -> Sequence[Project]:
+        return MetricsReleaseHealthBackend._get_projects_and_org_id(project_ids)[0]
+
+    @staticmethod
+    def _get_projects_and_org_id(project_ids: Sequence[int]) -> Tuple[Sequence[Project], int]:
+        projects = Project.objects.get_many_from_cache(project_ids)
+        org_ids: Set[int] = {project.organization_id for project in projects}
+        if len(org_ids) != 1:
+            raise ValueError("Expected projects to be from the same organization")
+
+        return projects, org_ids.pop()
+
+    @staticmethod
+    def _get_crash_free_rate_data(
+        org_id: int,
+        projects: Sequence[Project],
+        start: datetime,
+        end: datetime,
+        rollup: int,
+    ) -> Dict[int, Dict[str, float]]:
+
+        project_ids = [p.id for p in projects]
+
+        select = [
+            MetricField(metric_mri=SessionMRI.CRASHED.value, alias="crashed", op=None),
+            # named it 'init' to keep the same name as the original tag
+            MetricField(metric_mri=SessionMRI.ALL.value, alias="init", op=None),
+        ]
+
+        groupby = [
+            MetricGroupByField(field="project_id"),
+        ]
+        query = MetricsQuery(
+            org_id=org_id,
+            project_ids=project_ids,
+            select=select,
+            start=start,
+            end=end,
+            granularity=Granularity(rollup),
+            groupby=groupby,
+        )
+        result = get_series(projects=projects, metrics_query=query, use_case_id=USE_CASE_ID)
+
+        groups = get_path(result, "groups", default=[])
+        ret_val = {}
+        for group in groups:
+            project_id = get_path(group, "by", "project_id")
+            assert project_id is not None
+            totals = get_path(group, "totals")
+            assert totals is not None
+            ret_val[project_id] = totals
+
+        return ret_val
+
+    @staticmethod
+    def _compute_crash_free_rate(data: Dict[str, float]) -> Optional[float]:
+        total_session_count = data.get("init", 0)
+        crash_count = data.get("crashed", 0)
+
+        if total_session_count == 0:
```

Comment:
here you should be very careful, because if data have this key:
```
data = {"init": None}
```

you will get an `Exception`,  because your `default=0` will not work.

Reply:
Yep, true.
I don't know if we get values with `None`... will change to be defensive and override `None` with 0 in case we do.

Thanks.